keel-bakery-plugin/src/main/kotlin/com/netflix/spinnaker/keel/bakery/artifact/ImageHandler.kt

```diff
@@ -30,68 +30,87 @@ import org.springframework.context.ApplicationEventPublisher
 class ImageHandler(
   private val repository: KeelRepository,
   private val baseImageCache: BaseImageCache,
+  private val bakedImageRepository: BakedImageRepository,
   private val igorService: ArtifactService,
   private val imageService: ImageService,
-  private val diffFingerprintRepository: DiffFingerprintRepository,
   private val publisher: ApplicationEventPublisher,
   private val taskLauncher: TaskLauncher,
   private val defaultCredentials: BakeCredentials
 ) : ArtifactHandler {
 
   override suspend fun handle(artifact: DeliveryArtifact) {
     if (artifact is DebianArtifact) {
-      val latestArtifactVersion = try {
+      val desiredAppVersion = try {
         artifact.findLatestArtifactVersion()
       } catch (e: NoKnownArtifactVersions) {
         log.debug(e.message)
         return
       }
 
-      if (taskLauncher.correlatedTasksRunning(artifact.correlationId(latestArtifactVersion))) {
+      if (taskLauncher.correlatedTasksRunning(artifact.correlationId(desiredAppVersion))) {
         publisher.publishEvent(
           ArtifactCheckSkipped(artifact.type, artifact.name, "ActuationInProgress")
         )
       } else {
-        val latestBaseAmiName = artifact.findLatestBaseAmiName()
+        val desiredBaseAmiName = artifact.findLatestBaseAmiName()
 
-        val desired = Image(
-          baseAmiName = latestBaseAmiName,
-          appVersion = latestArtifactVersion,
-          regions = artifact.vmOptions.regions
-        )
-        val current = artifact.findLatestAmi()
-        val diff = DefaultResourceDiff(desired, current)
-
-        if (current != null && diff.isRegionsOnly()) {
-          if (current.regions.containsAll(desired.regions)) {
-            log.debug("Image for ${current.appVersion} contains more regions than we need, which is fine")
-          } else {
-            log.warn("Detected a diff in the regions for ${current.appVersion}: ${diff.toDebug()}")
-            publisher.publishEvent(ImageRegionMismatchDetected(current, artifact.vmOptions.regions))
+        val byArtifactVersion =
+          artifact.wasPreviouslyBakedWith(desiredAppVersion, desiredBaseAmiName)
+        if (byArtifactVersion) {
+          return
+        }
```

Comment:
Just nope out of this whole check if we know we've already done this. There's a possibility that River's cache has yet to update fully and we might unintentionally launch additional tasks that will just no-op (in fact that's been a long-standing gripe with this class).